src/tic_tac_toe/validation/move_validator.clj

```diff
@@ -0,0 +1,45 @@
+(ns tic-tac-toe.validation.move-validator
+  (:require [tic-tac-toe.board :as board]))
+
+(def invalid-spot "Not a valid spot.\n\n")
+(def out-of-range "Not a number between 1-%d.\n\n")
+(def spot-taken "Spot already taken.\n\n")
+
+(defn convert-to-index
+  [move]
+  (- (Integer/parseInt move) 1))
+
+(defn is-int?
+  [move]
+  (not= nil (re-matches #"\d+" move)))
+
+(defn in-range?
+  [board move]
+  (not= nil (some #(= move %) (range (count board)))))
+
+(defn available?
+  [board move]
+  (board/is-empty? (board move)))
+
+(defn get-error
+  [board move]
+  (cond
+    (not (in-range? board move)) (format out-of-range (count board))
+    (not (available? board move)) spot-taken))
+
+(defn valid-spot?
+  [board move]
+  (let [move (convert-to-index move)]
+    (if (and (in-range? board move) (available? board move))
+      {:valid true :data move}
+      {:valid false :data (get-error board move)})))
+
+(defn check-spot
+  [board move]
+  (cond
+    (not (string? move))
+      {:valid true :data move}
+    (is-int? move)
+      (valid-spot? board move)
+    :else
+      {:valid false :data invalid-spot}))
```

Comment:
Are all of these functions used outside of the ns? Have you considered making some of these private via `defn-`? Just curious.

Reply:
They are not used outside of the ns.  There are quite a few functions in different namespaces that I think should be private.  I debated a bit between public and testable vs. private, and left them public just to test more.  For these though, it would be easy enough to just test check-spot and make the rest private, so maybe I'll go ahead and do that.